spyder/plugins/editor/widgets/codeeditor.py

```diff
@@ -2872,6 +2883,63 @@ def run_pygments_highlighter(self):
         if isinstance(self.highlighter, sh.PygmentsSH):
             self.highlighter.make_charlist()
 
+    def left_line(self, position, source_button):

```

Comment:
I would like to see this split into two functions. 

One called `cursor_changed _lines` that checks if the cursor has changed lines and returns true. This could be useful to other code.

The second would could be called `strip_trailing_spaces` and this would be called first and `cursor_changed _lines` would be called from there.